Separates the src in modules

AdriCappelletti opened the pull request.

Commit:
Separates the src in modules

Commit:
Separate the code in modules

jabsatz commented on src/api.js:
```diff
@@ -1,9 +1,12 @@
     console.error(error);
-  });
+  }
+};
```
creo que acá estás confundiendo promesas comunes con async/await. Típicamente el try/catch se usa cuando tenés una función async porque eso te tira errores en este nivel, si estás usando una promesa común, tenés que usar `.catch()`. También, el último `.then()` es innecesario porque no hace ninguna transformación sobre responseJSON, simplemente la devuelve.
Después de esos cambios quedaría así:
```
export const fetchApi = (currentPage) => {
  return fetch(`https://pokeapi.co/api/v2/pokemon?limit=20&offset=${currentPage}`)
    .then((response) => response.json())
    .catch(error => console.error(error))
};
```

jabsatz commented on src/ui.js:
```diff
@@ -39,6 +46,24 @@ const createPokemonCardBack = ($pokemonCard, pokemonIndex) => {
   $cardBack.appendChild($weightParagraph);
 };
 
+export const goPrevPage = () => {
```
estos nombres de funciones me hacen un poco de ruido, porque al ejecutarlas no estás "yendo" a otra pagina por ejemplo, si no que estás asignando el event handler a un elemento. nombres mejores serían onda `assignPrevPageHandler` en mi opinión.

jabsatz commented on src/app.js:
```diff
@@ -1,51 +1,17 @@
-const $btnNext = document.querySelector('.btn-next');
-
-function appHandler() {
+async function appHandler() {
```
estas funciones no me convencen con los nombres, las funciones típicamente tienen que tener algún verbo para indicar que hacen. algo como `setupAppHandler` me gustaría más.

jabsatz reviewed (commented):
en temas de organización de proyecto no lo veo mal (aunque no estoy muy acostumbrado a revisar proyectos en javascript sin framework), lo que más veo que necesitás es claridad en los nombres y, capaz esto sea medio de quisquilloso, pero renombraría `utility.js` a `helpers.js` porque "helper" es el nombre que se usa en general para funciones cortitas de utilidad general.

jabsatz commented on src/ui.js:
```diff
@@ -0,0 +1,140 @@
+
+export const goPrevPage = () => {
+  $btnPrev.addEventListener('click', () => {
+    if (updatePrevPaginationInfo()) {
```
ah, esto sí que no me gusta. esta función está editando un contador, pero lo estás poniendo dentro de un if. en general, nunca deberías modificar cosas dentro de un if, pero además de eso no se entiende por el nombre de la función qué es lo que estás chequeando. viendo la implementación de la función, entiendo que es "si estás en una página mayor a 0", pero para eso me parecería mejor usar el clickCounter directamente acá:
```
if (clickCounter > 0) {
  updatePrevPagePaginationInfo()
  handlePokemonData();
}
```

fsodano commented on src/utility.js:
```diff
@@ -19,38 +20,31 @@ export const handlePokemonData = async () => {
+  } if (fetchedPokemons.includes(pokemonName)) {
+    return true;
+  }
+};
```
Acá estás procesando de más, estás recorriendo el array 2 veces, deberías guardar el resultado del `includes` en una variable y reutilizar

fsodano commented on src/utility.js:
```diff
@@ -19,38 +20,31 @@ export const handlePokemonData = async () => {
+export const updatePrevPaginationInfo = () => {
   if (clickCounter > 0) {
     clickCounter -= 1;
     currentPage -= 20;
```
por qué te vas 20 páginas para atrás?

fsodano commented on src/utility.js:
```diff
@@ -19,38 +20,31 @@ export const handlePokemonData = async () => {
+export const updateNextPaginationInfo = () => {
+  if (clickCounter < maxAmountOfpages && clickCounter >= 0) {
+    clickCounter += 1;
+    currentPage += 20;
```
creoq ue el concepto no es "página", las páginas son como las de un libro, estos son registros. Debería haber otra función que sepa cuántos registros hay por página y haga el cálculo

Commit:
Small refactor

Changes de name of a few functions and variables to make the code more
readable.
Also change the name of the "utility" module to another more apropiate
